### PR TITLE
Add support to download SVG locally

### PR DESCRIPTION
**Pull request by j0k3r**
We can't process a SVG using PHP (with GD) so instead we sanitize the SVG (to ensure it's a real SVG and not a bad input file) and then just copy the response from the server locally.

Using that method my only concern is about security. Is https://github.com/darylldoyle/svg-sanitizer safe enough to trust it?
What kind of other security check we can add?

Related https://github.com/wallabag/wallabag/issues/5328

### CODE REVIEW

**Comment by tcitworld** [comment]
`svg-sanitizer` excludes `<script>` and `<foreignObject>` and entity loader is disabled so the basics are here.

**Review by tcitworld — DISMISSED** [review]

File: src/Wallabag/CoreBundle/Helper/DownloadImages.php [hunk 1]
```diff
@@ -146,6 +147,26 @@ public function processSingleImage($entryId, $imagePath, $url, $relativePath = n
...
+            $sanitizer->minify(true);
+            $sanitizer->removeRemoteReferences(true);
+            $cleanSVG = $sanitizer->sanitize((string) $res->getBody());
+
+            // add an extra validation by checking about `<svg `
+            if (false === $cleanSVG || false === strpos($cleanSVG, '<svg ')) {
+                $this->logger->error('DownloadImages: Bad SVG given', ['path' => $imagePath]);
+
+                return false;
+            }
```
Comment (Kdecherf) [inline src/Wallabag/CoreBundle/Helper/DownloadImages.php:164]: Shouldn't we put a try-catch here, like for lines 171-177?

Comment (j0k3r) [reply src/Wallabag/CoreBundle/Helper/DownloadImages.php:164]: What other alternative we have, like in the catch from lines 171-177?
Or maybe just a try/catch with a `return false` in the catch?

Comment (Kdecherf) [reply src/Wallabag/CoreBundle/Helper/DownloadImages.php:164]: Yes, we could put `return false` in the catch

**Review by Kdecherf — APPROVED** [review]

**Pull request merged**